### PR TITLE
Clean up PutFile code in makeCommit slightly

### PR DESCRIPTION
**Pull request by msteffen**
After walking through this code with Adele and Kevin, I thought that it might be possible to simplify it a bit (in particular, the "BuildCommit case" comment was out-of date and misleading, so I wanted to fix that). This PR is my best effort to do that

### CODE REVIEW

**Comment by ysimonson** [comment]
The code looks good to me, but the tests seem to be failing legitimately

**Comment by msteffen** [comment]
> The code looks good to me, but the tests seem to be failing legitimately.
> 
> As an aside, I'd imagine there's a performance gain in reducing the amount of work done in an STM, as you've done here. Is that correct?

Yeah, they were, though I think I just fixed the issue. As far as taking RPCs out of the STM, that's my hope (that it improves performance a little). I'm always a little nervous that one of the reads depends on a key that has to be read inside the txn, but I read through the code a few times and I'm pretty confident it's safe

**Comment by msteffen** [comment]
Update: I had to put retrieving the parent tree back into the STM (but now there's a comment explaining why it's there). The issue is that we don't want to resolve `branch` outside the STM (in case another commit races with this one, and we need to re-run the callback with a different `branch` head), and we can't retrieve the parent commit until we resolve `branch`

File: src/server/pfs/server/driver.go [hunk 1]
```diff
@@ -458,8 +458,18 @@ func (d *driver) makeCommit(pachClient *client.APIClient, ID string, parent *pfs
...
+	// 1. PutFile has been called on a finished commit (records != nil), and
+	//    we want to apply 'records' to the parentCommit's HashTree, use that new
+	//    HashTree for this commit's filesystem, and then finish this commit
+	// 2. BuildCommit has been called by migration (treeRef != nil) and we
+	//    want a new, finished commit with the given treeRef
+	// - In either case, store this commit's HashTree in 'tree', so we have its
+	//   size, and store a pointer to the tree (in object store) in 'treeRef', to
+	//   put in newCommitInfo.Tree.
+  // - We also don't want to resolve 'branch' or 'parent.ID' (if it's a branch)
+  //   outside the txn below, so the 'PutFile' case is handled (by computing
```
Comment (ysimonson) [inline src/server/pfs/server/driver.go]: nit: fix mixed tabbing

File: src/server/pfs/server/driver.go [hunk 2]
```diff
@@ -558,35 +568,33 @@ func (d *driver) makeCommit(pachClient *client.APIClient, ID string, parent *pfs
...
 			newCommitInfo.SizeBytes = uint64(tree.FSSize())
 			newCommitInfo.Finished = now()
-		}
 
-		// Update the repo size if we're on the master branch and this is not
-		// an open commit
-		if !isStartCommit && branch == "master" {
-			repoInfo.SizeBytes = newCommitInfo.SizeBytes
+			// If we're updating the master branch, also set the repo size (persisted
+      // below, along with new branch)
```
Comment (ysimonson) [inline src/server/pfs/server/driver.go]: nit: fix mixed tabbing

File: src/server/pfs/server/driver.go (hunk 1 above)
Comment (msteffen) [reply src/server/pfs/server/driver.go]: Fixed; ran gofmt (which I just forgot to do before committing :sweat_smile:)

File: src/server/pfs/server/driver.go (hunk 2 above)
Comment (msteffen) [reply src/server/pfs/server/driver.go]: Fixed

File: src/server/pfs/server/driver.go [hunk 3]
```diff
@@ -558,35 +568,33 @@ func (d *driver) makeCommit(pachClient *client.APIClient, ID string, parent *pfs
...
 
-		// Update the repo size if we're on the master branch and this is not
-		// an open commit
-		if !isStartCommit && branch == "master" {
-			repoInfo.SizeBytes = newCommitInfo.SizeBytes
+			// If we're updating the master branch, also set the repo size (persisted
+			// below, along with new branch)
+			if branch == "master" {
+				repoInfo.SizeBytes = newCommitInfo.SizeBytes
+			}
```
Comment (ysimonson) [inline src/server/pfs/server/driver.go]: it looks like we're no longer checking if records is nil before updating the repoInfo. Is this correct?

File: src/server/pfs/server/driver.go [hunk 4]
```diff
@@ -558,35 +568,33 @@ func (d *driver) makeCommit(pachClient *client.APIClient, ID string, parent *pfs
...
+		} else {
+			if err := d.openCommits.ReadWrite(stm).Put(newCommit.ID, newCommit); err != nil {
+				return err
+			}
```
Comment (ysimonson) [inline src/server/pfs/server/driver.go]: Likewise with the above comment; this used to only hit if records was nil

Comment (msteffen) [reply src/server/pfs/server/driver.go]: I think it should still work that way. The indentation is getting hard to read in GitHub (at least for me), but from my editor, this section is basically:
```
		// 1. Write 'newCommit' to 'openCommits' collection OR
		// 2. Finish 'newCommit' (if treeRef != nil or records != nil); see
		//    "FinishCommit case" above)
		if treeRef != nil || records != nil {
			if records != nil {
				...
			}

			// If we're updating the master branch, also update the repo size (see
			// "Update repoInfo" below)
			if branch == "master" {
				repoInfo.SizeBytes = newCommitInfo.SizeBytes
			}
		} else {
			if err := d.openCommits.ReadWrite(stm).Put(newCommit.ID, newCommit); err != nil {
				return err
			}
		}
```

File: src/server/pfs/server/driver.go (hunk 3 above)
Comment (msteffen) [reply src/server/pfs/server/driver.go]: See below—I believe the code should still work this way

**Pull request merged**